tests/lib/c_lib/src/test_qsort.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2021 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <ztest.h>
+
+static int compare_ints(const void *a, const void *b)
+{
+	int aa = *(const int *)a;
+	int bb = *(const int *)b;
+
+	return (aa > bb) - (aa < bb);
+}
+
+void test_qsort(void)
+{
+	{
+		int actual_int[] = { 1, 3, 2 };
+		const int expect_int[] = { 1, 3, 2 };
+
+		qsort(actual_int + 1, 0, sizeof(int), compare_ints);
+		zassert_mem_equal(actual_int, expect_int, sizeof(int) * ARRAY_SIZE(expect_int),
```

Comment:
`sizeof` alone should be enough since it returns the number of bytes.
```suggestion
		zassert_mem_equal(actual_int, expect_int, sizeof(expect_int),
```